Fix/broken menu auctions

W3stside opened the pull request:
# Attempt 1 at MenuAuctions bug fix

1. made testing prod vs dev easier in local
2. SCSS/code clean
3. Fixed MenuAuctions grabbing prop from null Auction object
  a. Created `bothTheoClosed` var to represent edge case state

luarx committed:
Declare secure environment variables in Travis configuration

W3stside committed:
re-added back logic to remove wallet locked banner if unlocked mid app polling (#374)

W3stside committed:
Added ability change wallets + show account/wallet (#375)

W3stside committed:
added back in missing Status icons for wallet

Commit:
init commit for slow.trade rebranding

Commit:
remove static prop blue on home button

Commit:
add back static prop blue on home button

Commit:
fix spinner fixed position

Commit:
fix positioning of auction container and footer

Commit:
contentpage style + content updates

Commit:
mobile style updates

Commit:
misc changes

Commit:
misc changes

W3stside committed:
passed in network to menuWallet

Commit:
misc changes

Commit:
Merge branch 'develop-michelSEP27' of https://github.com/gnosis/dx-react into develop-michelSEP27

Commit:
misc changes

Commit:
misc mobile responsive changes

Commit:
misc mobile responsive changes

Commit:
misc mobile responsive changes

Commit:
misc mobile responsive changes

Commit:
misc mobile responsive changes

Commit:
misc mobile responsive changes

Commit:
misc mobile responsive changes

Commit:
misc mobile responsive changes

Commit:
misc mobile responsive changes

W3stside committed:
no blue class TextSquare buttonCTA

W3stside committed:
merged in up-to-date develop

W3stside committed:
made testing prod vs dev easier in local

W3stside committed:
fixed MenuAuctions

1. changed JS to match solidity math rules
2. added bothTheoClosed var to point state when 2 tokn pairs theo closed

W3stside committed:
removed comments/cleanup/ min-height menuauction scss fix

ghost commented:
Travis automatic deployment:
 https://pr390--dxreact.review.gnosisdev.com

ghost commented:
Travis automatic deployment:
 https://pr390--dxreact.review.gnosisdev.com

Velenir commented on src/api/index.ts:
```diff
@@ -473,7 +473,7 @@ export const getOutstandingVolume = async (
   ])
 
-  const outstandingVolume = sellVolume.mul(price[0]).div(price[1]).sub(buyVolume)
+  const outstandingVolume = price[1].gt(0) ? sellVolume.mul(price[0]).div(price[1]).sub(buyVolume) : toBigNumber(0)
```
Do a shortcut:
```
if (price[1].eq(0)) return toBigNumber(0)

const outstandingVolume = sellVolume.mul(price[0]).div(price[1]).sub(buyVolume)
```

Velenir commented on src/api/index.ts:
```diff
@@ -921,8 +923,9 @@ export const getSellerOngoingAuctions = async (
-      const currAuctionNeverRanDir = balanceNormal.eq(0) && closingPriceDir[1].eq(0)
-      const currAuctionNeverRanOpp = balanceInverse.eq(0) && closingPriceOpp[1].eq(0)
+      const bothAuctionTheoClosed = statusDir.theoreticallyClosed && statusOpp.theoreticallyClosed
+      const currAuctionNeverRanDir = bothAuctionTheoClosed || (balanceNormal.eq(0) && closingPriceDir[1].eq(0))
```
Direct auction had balance, which was breaking, or was it the opposite?

Velenir commented on src/api/contracts.ts:
```diff
@@ -49,8 +49,11 @@ interface ContractsMapWProxy extends ContractsMap {
 }
 
+// TODO: keep as 'development'
+const FE_CONDITIONAL_ENV = 'development'
```
what for?

W3stside replied:
testing without `npm run start:prod` and keeping logging

Velenir commented on src/api/index.ts:
```diff
@@ -754,6 +754,8 @@ const getLastAuctionStats = async (DutchX: DutchExchange, pair: TokenPair, accou
   else {
+    // outstandingVolume <= 0 === THEORETICALLY CLOSED
+    // TODO: ask why num here and not den
     if (closingPriceDir[0].equals(0) && outstandingVolumeNormal.eq(0)) {
```
doesn't matter, can be den

W3stside committed:
shortneded logic for outstandingvolume

ghost commented:
Travis automatic deployment:
 https://pr390--dxreact.review.gnosisdev.com

ghost commented:
Travis automatic deployment:
 https://pr390--dxreact.review.gnosisdev.com

W3stside committed:
fixed Auction Logic

W3stside committed:
uninstalled unused packages + created test ENV var

1. FE_CONDITIONAL_ENV used to test prod locally w/o build
2. removed await import as unused

W3stside committed:
passed in FE_CONDITIONAL_ENV var + cleanup

W3stside committed:
Merge remote-tracking branch 'origin/develop' into fix/broken-menu-auctions

W3stside committed:
update p-lock

ghost commented:
Travis automatic deployment:
 https://pr390--dxreact.review.gnosisdev.com

ghost commented:
Travis automatic deployment:
 https://pr390--dxreact.review.gnosisdev.com
